### PR TITLE
go API: Add support for more than one block device

### LINKED ISSUE #137: support multiple disks in Go API
We want to be able to support attaching multiple disks from the Go API for LinuxKit, currently we only do for qemu. 

This is needed to fix https://github.com/moby/tool/issues/68 and make all the tests more portable.

Plus it is just a useful feature.

### PR DESCRIPTION
**Pull request by MagnusS**
Changes DiskImage to an array called DiskImages with keys Path and
SizeMB. Also changes the JSON key from "disk" to "disk_images".
DiskSize is removed as size is set per disk image.

This allows multiple block devices to be mounted via the API and fixes #137

### CODE REVIEW

File: go/hyperkit.go [hunk 1]
```diff
@@ -70,6 +69,12 @@ type Socket9P struct {
 	Tag  string `json:"tag"`
 }
 
+// DiskImage contains the path to a disk image and an optional size if the image needs to be created.
+type DiskImage struct {
+	Path   string `json:"path"`
+	SizeMB int    `json:"size"`
```
Comment (rn) [inline go/hyperkit.go]: Can we call this `DiskConfig` and `Size` in the same way as https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/util.go#L139
The code is in different repos, but might as well try to keep it consistent

File: go/hyperkit.go [hunk 2]
```diff
@@ -82,8 +87,8 @@ type HyperKit struct {
 	VPNKitKey string `json:"vpnkit_key"`
 	// UUID is a string containing a UUID, it sets BIOS DMI UUID for the VM (as found in /sys/class/dmi/id/product_uuid on Linux).
 	UUID string `json:"uuid"`
-	// DiskImage is the path to the disk image to use
-	DiskImage string `json:"disk"`
+	// DiskImages contains disk images to use/create.
+	DiskImages []DiskImage `json:"disk_images"`
```
Comment (rn) [inline go/hyperkit.go]: Maybe make this: `Disks []DiskConfig` like here: https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/run_qemu.go#L27

File: go/sample/main.go [hunk 3]
```diff
@@ -33,7 +33,7 @@ func main() {
 	hk := flag.String("hyperkit", "", "HyperKit binary to use")
 	statedir := flag.String("state", "", "Directory to keep state in")
 	vpnkitsock := flag.String("vpnkitsock", "auto", "Path to VPNKit socket")
-	disk := flag.String("disk", "", "Path to disk image")
+	disks := flag.String("disk", "", "Comma separated list of disk images, format [image]{:size_in_MB}. If size is specified the image will be created if it doesn't exist. If image name is omitted a default name will be used.")
```
Comment (rn) [inline go/sample/main.go]: Could we maybe make this the same as/similar to the LinuxKit command line, as in https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/run_qemu.go#L64

this also has a more elgant way for parsing the command line in: https://github.com/linuxkit/linuxkit/blob/master/src/cmd/linuxkit/util.go#L153 so the code below can be moved into a method

Comment (MagnusS) [inline go/sample/main.go]: Yeah I just added something simple in the sample. But it's nice to be consistent - I'll update

**Comment by MagnusS** [comment]
@rn thanks for the review. I've updated to make it more consistent with current linuxkit.

File: go/sample/main.go [hunk 4]
```diff
@@ -101,9 +100,14 @@ func main() {
 		}
 		h.VSockPorts = ports
 	}
-	h.DiskImage = *disk
 	h.ISOImage = *iso
 
+	if len(disks) == 0 {
+		log.Fatalln("No disk specified")
```
Comment (justincormack) [inline go/sample/main.go]: Its fine to specify no disks surely?

**Comment by justincormack** [comment]
Minor comment but LGTM

File: go/sample/main.go (hunk 4 above)
Comment (MagnusS) [inline go/sample/main.go]: Yes - updated

**Pull request merged**